p2p/ipld/nmt_wrapper.go

```diff
@@ -16,18 +16,23 @@ var _ rsmt2d.TreeConstructorFn = ErasuredNamespacedMerkleTree{}.Constructor
 var _ rsmt2d.Tree = &ErasuredNamespacedMerkleTree{}
 
 // ErasuredNamespacedMerkleTree wraps NamespaceMerkleTree to conform to the
-// rsmt2d.Tree interface while catering specifically to erasure data. For the
-// first half of the tree, it uses the first DefaultNamespaceIDLen number of
-// bytes of the data pushed to determine the namespace. For the second half, it
-// uses the parity namespace ID
+// rsmt2d.Tree interface while also providing the correct namespaces to the
+// underlying NamespaceMerkleTree. It does this by adding the already included
+// namespace to the first half of the tree, and then uses the parity namespace
+// ID for each share pushed to the second half of the tree. This allows for the
+// namespaces to be included in the erasure data, while also keeping the nmt
+// library sufficiently general
 type ErasuredNamespacedMerkleTree struct {
 	squareSize uint64 // note: this refers to the width of the original square before erasure-coded
 	options    []nmt.Option
 	tree       *nmt.NamespacedMerkleTree
 }
 
-// NewErasuredNamespacedMerkleTree issues a new ErasuredNamespacedMerkleTree
+// NewErasuredNamespacedMerkleTree issues a new ErasuredNamespacedMerkleTree. Panics for squareSize of zero.
```

Comment:
Super nitpicky optional nit: I'm a friend of stating how this is to be used instead of saying when it panics. E.g. I'd write: "squareSize must be greater than zero" or "squareSize must be greater than zero or else it will panic".

Reply:
classy https://github.com/lazyledger/lazyledger-core/pull/306/commits/422c5732a87792984e15b21a239fcf0da7bad974